x-pack/plugins/security_solution/server/endpoint/routes/metadata/handlers.ts

```diff
@@ -125,33 +129,35 @@ export const getMetadataListRequestHandler = function (
         context.core.savedObjects.client
       );
 
-      body = await legacyListMetadataQuery(
-        context,
-        request,
-        endpointAppContext,
-        logger,
-        endpointPolicies
-      );
+      const pagingProperties = await getPagingProperties(request, endpointAppContext);
+
+      body = await legacyListMetadataQuery(context, endpointAppContext, logger, endpointPolicies, {
+        page: pagingProperties.pageIndex,
+        pageSize: pagingProperties.pageSize,
+        kuery: request?.body?.filters?.kql || '',
+        hostStatuses: request?.body?.filters?.host_status || [],
+      });
       return response.ok({ body });
     }
 
     // Unified index is installed and being used - perform search using new approach
     try {
       const pagingProperties = await getPagingProperties(request, endpointAppContext);
-      const { data, page, total, pageSize } = await endpointMetadataService.getHostMetadataList(
+      const { data, total } = await endpointMetadataService.getHostMetadataList(
         context.core.elasticsearch.client.asCurrentUser,
         {
-          page: pagingProperties.pageIndex + 1,
+          page: pagingProperties.pageIndex,
           pageSize: pagingProperties.pageSize,
-          filters: request.body?.filters || {},
+          hostStatuses: request.body?.filters.host_status || [],
+          kuery: request.body?.filters.kql || '',
         }
       );
 
       body = {
         hosts: data,
-        request_page_index: page - 1,
         total,
-        request_page_size: pageSize,
+        request_page_index: pagingProperties.pageIndex * pagingProperties.pageSize,
```

Comment:
the naming here is a bit confusing, this was actually the index of the document since it was being pulled directly from the `esQuery.from`. I changed everything to be index of the page so this is the last place where we're using index of doc, which we can then delete together with the route.

Reply:
😱 🙀  - So happy we are deleting this API.